cairo/ethereum/utils/numeric.cairo

```diff
@@ -148,3 +150,36 @@ func U256__eq__(a: U256, b: U256) -> bool {
     tempvar res = bool(0);
     return res;
 }
+
+// @dev Panics if overflow
+func U256_add{range_check_ptr}(a: U256, b: U256) -> U256 {
+    alloc_locals;
+    let (res, carry) = uint256_add(cast([a.value], Uint256), cast([b.value], Uint256));
+    if (carry != 0) {
+        with_attr error_message("OverflowError") {
+            assert 0 = 1;
+        }
+    }
+    tempvar result = U256(new U256Struct(res.low, res.high));
+    return result;
+}
+
+// @dev Panics if underflow with OverflowError
+func U256_sub{range_check_ptr}(a: U256, b: U256) -> U256 {
+    alloc_locals;
+    let is_within_bounds = U256_le(b, a);
+    if (is_within_bounds.value == 0) {
+        with_attr error_message("OverflowError") {
+            assert 0 = 1;
+        }
+    }
+    let (result) = uint256_sub(cast([a.value], Uint256), cast([b.value], Uint256));
+    tempvar res = U256(new U256Struct(result.low, result.high));
+    return res;
+}
+
+func U256_le{range_check_ptr}(a: U256, b: U256) -> bool {
+    let (result) = uint256_le(cast([a.value], Uint256), cast([b.value], Uint256));
```

Comment:
```suggestion
    let (result) = uint256_le([a.value], [b.value]);
```
You should not have to cast: a.value is a Uint256*

(to apply in all functions above)

Reply:
I thought it'd be clearer that we're using U256Struct as a Uint256 here by casting it, even though U256Struct is just a type alias for Uint256